### PR TITLE
Add full special attack notifier

### PR DESCRIPTION
**Pull request by robinwithes**
Added a option to idle notifier to notify you when your special attack energy hits a certain threshold.

### CODE REVIEW

**Comment by deathbeam** [comment]
I dont think this needs threshold, just notify at full? And also not sure if this should be part of idle notifier

**Comment by robinwithes** [comment]
> 
> 
> I dont think this needs threshold, just notify at full? And also not sure if this should be part of idle notifier

I only used the treshold to be more consistant with the other things in idle notifier, where would you prefer this if not in idle notifier? I think it fits well enough.

**Comment by Hydrox6** [comment]
Could be a cool idea to get the current weapon's special attack % and notify when it reaches that % (maybe when it reaches multiples of that percent, so I can for example see when I have 2 DWH/BGS specs available), as it seems that's what the threshold was meant to be for.

**Comment by forsco** [comment]
And what if you only want it for the spec weapon in your inventory? Thats why a manual threshold is nice

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 1]
```diff
@@ -363,6 +365,11 @@ public void onGameTick(GameTick event)
 		{
 			notifier.notify("[" + local.getName() + "] has low oxygen!");
 		}
+
+		if (checkSpecialAttack())
+		{
+			notifier.notify("[" + local.getName() + "] has reached his special attack threshold!");
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: `"[" + local.getName() + "] has regained special attack!"`

File: runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java [hunk 2]
```diff
@@ -411,6 +418,26 @@ private boolean checkLowHitpoints()
 		return false;
 	}
 
+	private boolean checkSpecialAttack()
+	{
+		double specialAttack = client.getVar(VarPlayer.SPECIAL_ATTACK_PERCENT) * 0.1;
+		if (config.getSpecAttackThreshold() <= 0)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/idlenotifier/IdleNotifierPlugin.java]: This can be before accessing the varp

**Comment by robinwithes** [comment]
Forgot this pr and i've seen someone else already did this.

**Pull request closed without merging**